### DISCUSSION TITLE
Mains switched Zigbee light bulbs, use detect_non_ha_changes or not?

### DISCUSSION THREAD

**Discussion by tbaumann** [basnijholt/adaptive-lighting#413] [Q&A]
I'm using zigbee lightbulbs that are switched via mains voltage switch.

It's not ideal, but lighting control is not something want to rewire in my rented apartment just for some adaptive lights.

So most of the time the light entries are actually "Unavailable" and when they switch on it takes a while. It mostly works all but I don't think the adaptive brightness really reacts very predictably to lights going on (online).
They eventually start flickering around (I think mostly because the zigbee tuya lights are garbage) and eventually settle on the right colourtemperature.

Question is, is this switching on and off by mains switch considered external control? Should I use `detect_non_ha_changes`.

**Comment by th3w1zard1** [comment]
No, config options like `take_over_control` and `detect_non_ha_changes` are for users who want to set their light's brightness/color_temp/rgb outside of HASS, if you're simply turning them on and off, and you want them to adapt while they're on, you're good with the default settings :)
